### DISCUSSION TITLE
bug in step reported in mem.trace

### DISCUSSION THREAD

**Discussion by Soilorian** [sarchlab/mgpusim#92] [Q&A]
Hello,

I believe I've found another bug in the simulator. When you run a benchmark with the `-trace-mem` flag and examine the 'step' reports, some of them don't have the correct IDs like the other reports, more specifically L2 IDs are correct but L1 IDs contain only a number, whereas they should include a number followed by `@` and the source of the request.

I have also reviewed the code and found a solution. If you agree that this is a bug, I can submit a pull request.

Best regards,  
Mahdi,
Sharif University of Technology

**Comment by Soilorian** [comment]
I have inspected this problem more and noticed that the change of ID is because of the colaser and by colassing different requests it creates a new ID for them, I have added a list of colassed IDs to transaction so I can keep track of all the IDs that are being colassed and print them on each step

**Reply by syifan** [reply]
I am not 100% if I understand what you mean. If you think it is an error and have attempted to fix it, could you start a pull request?

**Reply by Soilorian** [reply]
maybe I didn't state my thoughts correctly,
no it's not a bug so I won't start a pull request,
we just needed to track the transactions after going to L2 and we thought that the ids were changing but that was not the case.
